virttest/utils_test/__init__.py

```diff
@@ -1947,149 +1956,147 @@ def _tarball_download(self, url, destination):
         self.base_name = tarball.members[0].name.split("/")[0]
         tarball.close()
 
-    @error_context.context_aware
     def download_stress(self):
         """
         Download stress tool
         """
         url = self.download_url
         tmp_path = data_dir.get_tmp_dir()
-        error_context.context('Download stress tool from %s' % url,
-                              logging.info)
+        logging.info('Download stress tool from %s', url)
         download_method = getattr(self, "_%s_download" % self.download_type)
         download_method(url, tmp_path)
-        error_context.context('Copy stress tool to guest', logging.info)
-        self.vm.wait_for_login()
-        self.vm.copy_files_to(os.path.join(tmp_path, self.base_name), self.dst_path)
+        source = os.path.join(tmp_path, self.base_name)
+        if self.remote_host:
+            logging.info('Copy stress tool to remote host')
+
+            args = (self.remote_host.__getitem__('server_ip'), 'scp',
+                    self.remote_host.__getitem__('server_user'),
+                    self.remote_host.__getitem__('server_pwd'), "22",
+                    source, self.dst_path)
+            self.copy_files_to(args)
+        else:
+            logging.info('Copy stress tool to work dir of host/guest')
+            self.copy_files_to(source, self.dst_path)
 
-    @error_context.context_aware
     def install(self):
```

Comment:
i'm not sure if we needed @session_handler() before def install(), for it runs in guest.

Reply:
@balamuruhans, i think there may be some problem if  i call install() directly in https://github.com/autotest/tp-qemu/pull/1388, for there is no session defined in install(). Any suggestion?

Reply:
I believe it is still possible to use `install()` directly from your patch by something like,
```python
from virttest import utils_test
stress_obj = utils_test.Stress(**args)
stress_obj.session = vm.wait_for_login()
stress_obj.install()
```

Reply:
OK, then i'll define a new session in my .py, thank you.

Reply:
@balamuruhans and @vivianQizhu, i'm clear now. self.session needed is initialized in VMStress.__init__(), so i can call install() in my *.py directly.
 like, 
  ****iozone.py******
   stress_obj = utils_test.VMStress(**args)
   stress_obj.install()

Thanks for all your help.